### ISSUE TITLE
Explore 'vaadin-split-layout'

### ISSUE THREAD

**Issue by anoblet** [conversionxl/aybolit#83]
Work can be seen here: https://github.com/anoblet/aybolit/tree/vaadin-split-layout

**Comment by lkraav** [comment]
@anoblet post a sneak peek screenshot or a short screencap?

**Comment by anoblet** [comment]
https://www.youtube.com/watch?v=StJZ4HwzKOc&feature=youtu.be

**Comment by anoblet** [comment]
@lkraav I'm writing it in Typescript as that is what I'm familiar with. Trying to resolve two issues with one PR :) Typescript/Babel is proving to be a little bit difficult though.

**Comment by lkraav** [comment]
My original idea was to try to refactor `<cxl-app-layout>` directly, which provides ability to verify result in existing stories much easier.

But it may be hard to grasp all the details there, we could do a introductory screenshare session.

I just pushed a RC to https://github.com/conversionxl/aybolit/tree/leho-cxl-app-layout definitely rebase to this.

**Comment by anoblet** [comment]
I intend on bringing in the site specific content (header, navigation, main, footer) soon. My intent was to isolate the functionality of `vaadin-split-layout`.

I'm open to a screen share session :)

**Comment by lkraav** [comment]
> I intend on bringing in the site specific content (header, navigation, main, footer) soon. My intent is to isolate the functionality of `vaadin-split-layout`.

Yes, but this creates a relatively large re-implementation of everything 1-column.

Instead, you could patch cxl-app-layout's current template, maybe drop most "2c" css, and start from that point.

You'd have a significant head start towards meeting end result expectations.

**Comment by anoblet** [comment]
The current iteration relies on `position`. If position is not defined it defers to one column. `position="left"` uses `vaadin-split-layout` with the `aside` column to the left. `position="right"` places the aside column to the right. If `wide` is false or the user is on mobile, the `aside` is placed before the content and the orientation is set to vertical.

**Comment by anoblet** [comment]
I updated the branch to be refactored based on cxl-app-layout. Looks like most it works apart from some styling of the side bar. Screen capture can be seen here: https://youtu.be/9TQxl7LYo44
